ceremonies/meetup-validation/src/tests.rs

```diff
@@ -0,0 +1,228 @@
+		UpdatedParticipants { included: participants.clone(), excluded: vec![] };
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_1() {
+	// Scenario: One participant did not vote
+	let participants: Vec<usize> = vec![0, 1, 2, 3];
+	let participant_votes: Vec<u32> = vec![0, 3, 3, 3];
+	let participant_attestations: Vec<Vec<usize>> =
+		vec![vec![1, 2, 3], vec![0, 2, 3], vec![0, 1, 3], vec![0, 1, 2]];
+
+	let updated_participants = UpdatedParticipants {
+		included: vec![1, 2, 3],
+		excluded: vec![ExcludedParticipant { index: 0, reason: ExclusionReason::NoVote }],
+	};
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_2() {
+	// Scenario: One participant did not vote like majority
+	let participants: Vec<usize> = vec![0, 1, 2, 3];
+	let participant_votes: Vec<u32> = vec![1, 3, 3, 3];
+	let participant_attestations: Vec<Vec<usize>> =
+		vec![vec![1, 2, 3], vec![0, 2, 3], vec![0, 1, 3], vec![0, 1, 2]];
+
+	let updated_participants = UpdatedParticipants {
+		included: vec![1, 2, 3],
+		excluded: vec![ExcludedParticipant { index: 0, reason: ExclusionReason::WrongVote }],
+	};
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_3() {
+	// Scenario: 5 participants
+	// 0 has a broken phone and can only be attested by 1 and 2 and attests nobody
+	// we expect that everybody except for 0 gets a reward
+	let participants: Vec<usize> = vec![0, 1, 2, 3, 4];
+	let participant_votes: Vec<u32> = vec![5, 5, 5, 5, 5];
+	let participant_attestations: Vec<Vec<usize>> =
+		vec![vec![], vec![0, 2, 3, 4], vec![0, 1, 3, 4], vec![1, 2, 4], vec![1, 2, 3]];
+
+	let updated_participants = UpdatedParticipants {
+		included: vec![1, 2, 3, 4],
+		excluded: vec![ExcludedParticipant {
+			index: 0,
+			reason: ExclusionReason::TooFewOutgoingAttestations,
+		}],
+	};
+	let attestation_threshold_fn = |i| i - 1;
+
+	assert_eq!(
+		get_updated_participants(
+			&participants,
+			&participant_votes,
+			&participant_attestations,
+			attestation_threshold_fn,
+		)
+		.unwrap(),
+		updated_participants
+	);
+}
+
+#[test]
+fn get_updated_participants_works_case_4() {
+	// Scenario: 5 participants
+	// 0 has a broken screen and can only be attested by 1 and 2 but attests everybody else
+	// we expect that everybody except for 0 gets a reward
```

Comment:
so here we have the actual improvement in behaviour over the status quo: This mitigates the problem that someone with ill intent (or good intent but bad gear) can harm others